packages/react/src/FilteredActionList/FilteredActionList.tsx

```diff
@@ -44,6 +46,19 @@ export interface FilteredActionListProps extends Partial<Omit<GroupedListProps,
   announcementsEnabled?: boolean
   fullScreenOnNarrow?: boolean
   onSelectAllChange?: (checked: boolean) => void
+  /**
+   * Enable virtualization for large lists to improve rendering performance.
+   * Only renders visible items plus a small overscan buffer, reducing DOM node count and improving scroll performance.
+   *
+   * When to use: Enable this option when rendering lists with many items to avoid performance bottlenecks.
+   *
+   * Limitations:
+   * - Virtualization is only supported for non-grouped ActionLists.
+   * - For best results, items should have consistent heights.
+   *
+   * @default false
+   */
+  isVirtualized?: boolean
```

Comment:
Is there any way to structure this so it's more of a "bolted-on" addition rather than embedded directly within `FilteredActionList`? 

Nothing against the code you have here (and not a blocker for you to consider), I'm just a little hesitant when adding a features like this that require fundamental changes to the core component, as it's typically a sign that we aren't exposing the right primitive/modules to enable building out more complex interactions. Would love any insights from @joshblack or others as well

Reply:
I was thinking add the prop would be the safest way to test this functionality out -- I am hesitant to add this to ALL `FilteredActionList`'s since we have a few use cases attempting to self resolve this issue.

Reply:
💯 I'm definitely in favor of isolating the changes to avoid breaking anything unintentionally. I was more so thinking of the potential to do it as an independant utility: 

```tsx
const { filteredActionListProps } = useVirtualizedActionList({
   ...
   ...
})

return <FilteredActionList {...filteredActionListProps} />
``` 

but that may not be possible with the props or primitives exposed from FilteredActionList today